Ark.Tools.ResourceWatcher/ResourceWatcherDiagnosticSource.cs

```diff
@@ -189,17 +189,17 @@ public void CheckStateSuccessful(Activity activity, IEnumerable<ProcessContext>
         #region ProcessResource
         public Activity ProcessResourceStart(ProcessContext processContext)
         {
-            Tuple<string, LocalDateTime?, LocalDateTime?> infos;
-            bool result = processContext.IsNewResource(out infos);
+            (string source, LocalDateTime? current, LocalDateTime? last)? infos;
+            bool result = processContext.IsResourceUpdated(out infos);
```

Comment:
you can use 
`
processContext.IsResourceUpdated(out var infos);
`
Without declaring the var before.

Also, I think the `out ` parameters is not needed to be nullable. it's guaranteed to be not null if result is 'true' right?